irony.el

```diff
@@ -661,6 +661,17 @@ If no such file exists on the filesystem the special file '-' is
                              (format "%s\n"
                                      (combine-and-quote-strings argv)))))))
 
+(defun irony--send-request-sync (request callback &rest args)
+  "Send a request to irony-server and wait for the result.
+
+callback should set the irony-sync-result property of the irony-server process"
+  (let ((process (irony--get-server-process-create)))
+    ;; Using apply to "spread" the args list
+    (apply 'irony--send-request request callback args)
+    ;; While there are callbacks left on the queue, wait for output from server
+    (while (process-get process 'irony-callback-stack)
```

Comment:
You should make a function of it's own for the check of the callback stack, `irony--server-process-has-callback-p` or something along these lines, it's not impossible that I rework the callback stack mechanism a little bit, that would make it easier for me if it's already along with the other callback-stack manipulation functions.